src/services/events/index.ts

```diff
@@ -0,0 +1,123 @@
+import { CartDto, Product, User } from '../../interfaces'
+
+type EventCallbackFunction<Payload = any> = (payload: Payload) => void
+
+type GetTypeFromInterface<T> = T extends EventCallbackFunction<infer Payload> ? Payload : undefined
+
+type EventsListenerService = () => {
```

Comment:
Niech to jest nazwa typu który jest już obiektem, a nie fabryki tego obiektu bo to będzie bardziej intuicyjne

Czyli po prostu `interface EventsListenerService { emit: ... }`

Reply:
Druga sprawa, niej typ jest eksportowany i nazwany raczej jako `HeseyaEventListenerService`